Adding support for HTTPServletRequest attributes

johannnallathamby opened the pull request.

mefarazath commented on components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityRequest.java:
```diff
@@ -283,6 +285,31 @@ public IdentityRequestBuilder addParameters(Map<String, String[]> parameters) {
         }
 
+        public IdentityRequestBuilder setAttributes(Map<String, Object> attributes) {
+            this.attributes = new HashMap(attributes);;
```
Do we need to create a new HashMap here?

harsha1979 replied:
We need to revert this back and need to have HashMap to copy the original hashmap from servlet request to the identity request

mefarazath replied:
Let's do this copying at the Factory level since the bean need not be aware of the endpoint details

johannnallathamby commented:
I didn't want to allow the instance variable becoming null. I guess it
would be better to check for null and assign instead of creating a new
object.

On Fri, Dec 2, 2016 at 12:07 PM, Farasath Ahamed <notifications@github.com>
wrote:

> *@mefarazath* commented on this pull request.
> ------------------------------
>
> In components/authentication-framework/org.wso2.carbon.
> identity.application.authentication.framework/src/
> main/java/org/wso2/carbon/identity/application/authentication/framework/
> inbound/IdentityRequest.java
> <https://github.com/wso2/carbon-identity-framework/pull/701#pullrequestreview-11112379>
> :
>
> > @@ -283,6 +285,31 @@ public IdentityRequestBuilder addParameters(Map<String, String[]> parameters) {
>              return this;
>          }
>
> +        public IdentityRequestBuilder setAttributes(Map<String, Object> attributes) {
> +            this.attributes = new HashMap(attributes);;
>
> Do we need to create a new HashMap here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wso2/carbon-identity-framework/pull/701#pullrequestreview-11112379>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHa5Err8e8A60MHSfMxOSW1Ddlv8kP8cks5rD7yxgaJpZM4K89vh>
> .
>



-- 
Thanks & Regards,

*Johann Dilantha Nallathamby*
Technical Lead & Product Lead of WSO2 Identity Server
Governance Technologies Team
WSO2, Inc.
lean.enterprise.middleware

Mobile - *+94777776950*
Blog - *http://nallaa.wordpress.com <http://nallaa.wordpress.com>*

mefarazath commented on components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityRequest.java:
```diff
@@ -283,6 +285,31 @@ public IdentityRequestBuilder addParameters(Map<String, String[]> parameters) {
+
+        public IdentityRequestBuilder addAttributes(Map<String, Object> attributes) {
+            for (Map.Entry<String, Object> attribute : attributes.entrySet()) {
+                if (this.attributes.containsKey(attribute.getKey())) {
```
shall we replace 304-308 with `addAttribute(attribute.getKey(), attribute.getValue())` ?

johannnallathamby commented:
+1.

On Fri, Dec 2, 2016 at 12:23 PM, Farasath Ahamed <notifications@github.com>
wrote:

> *@mefarazath* commented on this pull request.
> ------------------------------
>
> In components/authentication-framework/org.wso2.carbon.
> identity.application.authentication.framework/src/
> main/java/org/wso2/carbon/identity/application/authentication/framework/
> inbound/IdentityRequest.java
> <https://github.com/wso2/carbon-identity-framework/pull/701#pullrequestreview-11113407>
> :
>
> > +            this.attributes = new HashMap(attributes);;
> +            return this;
> +        }
> +
> +        public IdentityRequestBuilder addAttribute(String name, Object value) {
> +            if (this.attributes.containsKey(name)) {
> +                throw FrameworkRuntimeException.error("Attributes map trying to override existing " +
> +                                                      "key " + name);
> +            }
> +            this.attributes.put(name, value);
> +            return this;
> +        }
> +
> +        public IdentityRequestBuilder addAttributes(Map<String, Object> attributes) {
> +            for (Map.Entry<String, Object> attribute : attributes.entrySet()) {
> +                if (this.attributes.containsKey(attribute.getKey())) {
>
> shall we replace 304-308 with addAttribute(attribute.getKey(),
> attribute.getValue()) ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wso2/carbon-identity-framework/pull/701#pullrequestreview-11113407>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHa5Et_wALLP-YhJPJs-Z6M6Sc2mOhLXks5rD8B5gaJpZM4K89vh>
> .
>



-- 
Thanks & Regards,

*Johann Dilantha Nallathamby*
Technical Lead & Product Lead of WSO2 Identity Server
Governance Technologies Team
WSO2, Inc.
lean.enterprise.middleware

Mobile - *+94777776950*
Blog - *http://nallaa.wordpress.com <http://nallaa.wordpress.com>*

johannnallathamby commented:
Fixed and updated the PR.

On Fri, Dec 2, 2016 at 12:25 PM, Johann Nallathamby <johann@wso2.com> wrote:

> +1.
>
> On Fri, Dec 2, 2016 at 12:23 PM, Farasath Ahamed <notifications@github.com
> > wrote:
>
>> *@mefarazath* commented on this pull request.
>> ------------------------------
>>
>> In components/authentication-framework/org.wso2.carbon.identity
>> .application.authentication.framework/src/main/java/org/
>> wso2/carbon/identity/application/authentication/framework/inbound/
>> IdentityRequest.java
>> <https://github.com/wso2/carbon-identity-framework/pull/701#pullrequestreview-11113407>
>> :
>>
>> > +            this.attributes = new HashMap(attributes);;
>> +            return this;
>> +        }
>> +
>> +        public IdentityRequestBuilder addAttribute(String name, Object value) {
>> +            if (this.attributes.containsKey(name)) {
>> +                throw FrameworkRuntimeException.error("Attributes map trying to override existing " +
>> +                                                      "key " + name);
>> +            }
>> +            this.attributes.put(name, value);
>> +            return this;
>> +        }
>> +
>> +        public IdentityRequestBuilder addAttributes(Map<String, Object> attributes) {
>> +            for (Map.Entry<String, Object> attribute : attributes.entrySet()) {
>> +                if (this.attributes.containsKey(attribute.getKey())) {
>>
>> shall we replace 304-308 with addAttribute(attribute.getKey(),
>> attribute.getValue()) ?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/wso2/carbon-identity-framework/pull/701#pullrequestreview-11113407>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AHa5Et_wALLP-YhJPJs-Z6M6Sc2mOhLXks5rD8B5gaJpZM4K89vh>
>> .
>>
>
>
>
> --
> Thanks & Regards,
>
> *Johann Dilantha Nallathamby*
> Technical Lead & Product Lead of WSO2 Identity Server
> Governance Technologies Team
> WSO2, Inc.
> lean.enterprise.middleware
>
> Mobile - *+94777776950*
> Blog - *http://nallaa.wordpress.com <http://nallaa.wordpress.com>*
>



-- 
Thanks & Regards,

*Johann Dilantha Nallathamby*
Technical Lead & Product Lead of WSO2 Identity Server
Governance Technologies Team
WSO2, Inc.
lean.enterprise.middleware

Mobile - *+94777776950*
Blog - *http://nallaa.wordpress.com <http://nallaa.wordpress.com>*
